app/main.py

```diff
@@ -2,0 +1,34 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return "{"\
+               f"Name: {self.name}, " \
+               f"Health: {self.health}," \
+               f" Hidden: {self.hidden}"\
+               "}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
```

Comment:
The `hide` method toggles the `hidden` state of a `Herbivore`. Consider renaming the method to something like `toggle_hidden` for clarity, as `hide` might imply only setting the state to hidden.